neuro_san/internals/run_context/langchain/token_counting/langchain_token_counter.py

```diff
@@ -118,7 +121,11 @@ async def count_tokens(self, awaitable: Awaitable) -> Any:
             # and use the create_task() to run within that context.
             new_context: Context = copy_context()
             task: Task = new_context.run(self.create_task, awaitable)
-            retval = await task
+            try:
+                retval = await wait_for(task, max_execution_seconds)
+            except AsyncTimeout:
+                # Per docs for wait_for(), the task is already cancelled.
+                retval = None
```

Comment:
Use the max_execution_seconds that is passed in as an arg here as a timeout.
This is the new recommendation per:
https://python.langchain.com/docs/how_to/migrate_agent/#in-langgraph-4